src/analyzing/AST_to_IL.ml

```diff
@@ -270,6 +270,22 @@ let def_expr_evaluates_to_value (lang : Lang.t) =
   | Elixir -> true
   | _else_ -> false
 
+let is_constructor env ret_ty id_info =
+  match id_info.G.id_resolved.contents with
+  | Some (G.GlobalName (ls, _), _) -> (
+      (* TODO this condition is a little too loose, but should work in
```

Comment:
This comment is very confusing, why is it "too loose", how could we "tighten this up" ???